test/e2e/apimachinery/resource_quota.go

```diff
@@ -1194,7 +1194,7 @@ var _ = SIGDescribe("ResourceQuota", func() {
 		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 6*time.Minute, true, func(ctx context.Context) (bool, error) {
 			resourceQuotaResult, err := rqClient.Get(ctx, rqName, metav1.GetOptions{})
 			if err != nil {
-				return false, nil
+				return false, err
 			}
 
 			if *resourceQuotaResult.Spec.Hard.Cpu() == *resourceQuotaResult.Status.Hard.Cpu() {
```

Comment:
we should revert this as per @liggitt 's comment and change other places to use Quantity#Equal  too.
it guarantees that if something changes in the struct we don't need to update all places.
...and if it happens static checks will not catch it.

Reply:
is not really revert, before was doing a deepEqual comparison, in this case AFAIK this can not happen since the test creates and checks the quota value, and was using this comparison for several years without issues

Reply:
for correctness all places must use the specialized equality functions.
other places should be changed too, ideally.

this thread talked more about it:
https://github.com/kubernetes/kubernetes/pull/121951/files#r1399312087

Reply:
yeah, but is not my intention to fix that , it was already there, I want to fix the thing that I have changed, that is that before it failed on unexpected apiserver conditions, and now it swallow